Adds adjacency map computation to the mesher

Rohit-Kakodkar opened the pull request:
## Description

- [x] Adds adjacency map computation
- [x] Writes adjacency map to database
- [x] Removes older implementation

## Issue Number

If there is an issue created for these changes, link it here

## Checklist

Please make sure to check developer documentation on specfem docs.

- [ ] I ran the code through pre-commit to check style
- [ ] **THE DOCUMENTATION BUILDS WITHOUT WARNINGS/ERRORS**
- [ ] I have added labels to the PR (see right hand side of the PR page)
- [ ] My code passes all the integration tests
- [ ] I have added sufficient unittests to test my changes
- [ ] I have added/updated documentation for the changes I am proposing
- [ ] I have updated CMakeLists to ensure my code builds
- [ ] My code builds across all platforms

Rohit-Kakodkar committed:
Adds adjacency map MESHFEM2D

- [x] Adds adjacency map computation
- [x] Writes adjacency map to database

Rohit-Kakodkar committed:
Fixed minor bugs

- Wrong index range for adjacency graph
- Edges double counted

Rohit-Kakodkar committed:
Merge branch 'devel-dg' into meshfem2d-adjacency-map

Rohit-Kakodkar committed:
Removed files | updated writer

Rohit-Kakodkar committed:
Set default value of adjacency map = true

Rohit-Kakodkar committed:
Solver adjacency map

- [x] Reads adjacency map generated by the mesher
- [x] Creates a boost graph for adjacency map with required properties

Rohit-Kakodkar committed:
reverted dockerfile

Rohit-Kakodkar committed:
Compute index mapping from adjacency graph

- [x] Adds a new assemble function
- [x] Creates coordinate mappings for connected elements

Rohit-Kakodkar committed:
Review changes

- [x] Merge vertex and edge enums for strongly conforming
- [x] Moved orientation to mesh_entities

Rohit-Kakodkar committed:
Updated boundaries enums to mesh_entities

Rohit-Kakodkar committed:
Merge branch 'issue-1071' into issue-1086

Rohit-Kakodkar committed:
Fixed issues with adjacency graph

- [x] Fixes issues with corner node numbering update
- [x] Adds check to ensure graph symmetry

Rohit-Kakodkar committed:
Merge pull request #1074 from PrincetonUniversity/issue-1071

Adjacency map for the solver

Rohit-Kakodkar commented:
@int-ptr-ptr Can you have a look at this?

Rohit-Kakodkar committed:
Fixed index mapping | working code

- [x] Compute index mapping from adjacency graph
- [x] Fixed issues with corner nodes
- [x] Fixed ordering of edges in mesh entities

Rohit-Kakodkar committed:
Added comments

Rohit-Kakodkar committed:
Fixes unity build

Rohit-Kakodkar committed:
Reduced parallel works in github checks

Rohit-Kakodkar committed:
Moved adjacency graph checks to mesh

Rohit-Kakodkar committed:
Adds tests for adjacency graph

- [x] Adds tests to check member functions within adjacency graph
- [x] Adds test to check the correctness of regular mesh
- [x] Adds test to check the correctness of circular mesh

Rohit-Kakodkar committed:
Fixed the jou file

Rohit-Kakodkar committed:
Merge pull request #1090 from PrincetonUniversity/issue-1086

Index mapping computation using adjacency graph

int-ptr-ptr commented on fortran/meshfem2d/meshfem2D.F90:
```diff
@@ -638,6 +639,13 @@ program meshfem2D
       if (output_grid_Gnuplot .and. .not. read_external_mesh) &
          call save_gnuplot_file(NGNOD,nx_elem_internal,nz_elem_internal,grid_point_x,grid_point_z)
 
+      if (read_external_mesh) then
+         ! @Kentaro Here you should be calling routine to read in adjacency information generated by external mesher
+      else
+         ! computes adjacency information for internal mesh
+         call compute_adjacency_graph()
+      endif
```
Would you want the adjacency information to be in the same format as the database output (minus binary formatting)?

```fortran
!! Write out adjacency graph
! i = current element
! adjacent_elements(i,j) = adjacent element number
! adjacency_type(i,j) = type of adjacency (1 = STRONGLY_CONFORMING)
! adjacency_id(i,j) = defines which edge/vertex is connected from i to adjacent_elements(i,j)
write(IOUT) i + 1, adjacent_elements(i,j) + 1, adjacency_type(i,j), adjacency_id(i,j)
```

Rohit-Kakodkar replied:
I updated this in the test PR. I realized the method for computing conforming adjacencies would be valid for any mesh, even for non-conforming ones. So we wouldn't need this information from the mesher anymore. See #1096

Rohit-Kakodkar replied:
I am also fine with writing the conforming adjacencies from CUBIT/GMESH as well if that's more appropriate.

Rohit-Kakodkar committed:
Merge pull request #1096 from PrincetonUniversity/issue-1095

Adds tests for adjacency graph

Rohit-Kakodkar committed:
Deprecates Adjacency map

- [x] Deprecates adjacency map
- [x] Adds displacement tests to check index mapping

Rohit-Kakodkar committed:
Adds Elastic Domain w Adjacency mapping

Rohit-Kakodkar committed:
Updated specfem_config.yaml

Rohit-Kakodkar committed:
Merge pull request #1111 from PrincetonUniversity/issue-1110

Deprecate adjacency map
